modules/scrabble.js

```diff
@@ -27,21 +27,31 @@ function scoreLetter (letter, str) {
     'y': { count: 2, score: 4 },
     'z': { count: 1, score: 10 }
   }
-  const occurrences = str.toLowerCase().split(letter).length - 1
-  const usedBlanks = Math.max(occurrences - bag[letter]['count'], 0)
-  return {
-    score: bag[letter]['score'] * (occurrences - usedBlanks),
-    usedBlanks
+  try {
+    const occurrences = str.toLowerCase().split(letter).length - 1
+    const usedBlanks = Math.max(occurrences - bag[letter]['count'], 0)
+    return {
+      score: bag[letter]['score'] * (occurrences - usedBlanks),
+      usedBlanks
+    }
+  } catch (e) {
+    if (e instanceof TypeError) {
+      return {
+        score: 0,
+        usedBlanks: 0
+      }
+    }
   }
 }
 function getPunctuation (bot, score) {
   const minScore = bot.config.get('scrabble.minScore')
   const exclThreshold = bot.config.get('scrabble.exclamationThreshold')
   const numberOfMarks = Math.floor((score - minScore) / exclThreshold)
-  return '!'.repeat(numberOfMarks || '.')
+  return '!'.repeat(numberOfMarks) || '.'
 }
 function scrabbleScore (str) {
-  if (str.length > 15 || str.length < 2 || /[^a-zA-Z]/.test(str)) {
+  if (str.length > 15 || str.length < 2 ||
+      /[^a-zA-Z]/.test(str.substring(0, str.length - 1))) {
```

Comment:
This won't match, say, `jukebox?!`. It might be better to extract the word from the cruft in the `message` event handler and only pass sensible inputs to the `scrabbleScore` function. That way it'll still reject `!scrabble jukebox!` as expected